firmware/controllers/engine_cycle/knock_controller.cpp

```diff
@@ -8,10 +8,14 @@
 #include "pch.h"
 #include "knock_logic.h"
 
+void KnockController::init(engine_configuration_s const * const previousConfig) {
```

Comment:
what is the point of 'previousConfig' here?

Reply:
you are correct, I was moving too fast, didn't realize `config` was "global" and `previousConfig` is not what `onConfigurationChange` was/is using